### PR TITLE
[Issue/3217] Migration flow improvements

### LINKED ISSUE #3217: USAM - Migration flow Improvements
1. confirmation popup before running migration
2. assembly references
3. we need a migration blog post doc-post

### PR DESCRIPTION
**Pull request by gabrielbeamable**
# Ticket

resolves #3217 

# Brief Description

This PR does:

- Goes through all old microservices/storages and not only the ones existing in the default folder
- Get all assembly references from the old microservices and save those in the new signpost file
- Creates a migration confirmation popup that will show if there are any services that need migration
- Fixes an issue with old services files and folders not being deleted

# Notes

When you are merging a feature branch into `main`, please squash merge and make sure the final commit contains any relevent JIRA ticket number. If you are merging from `main` to `staging`, or `staging` to `production`, please use a regular merge commit. 

Does this introduce tech-debt? If so, have you added an entry to the [Tech-debt document?](https://docs.google.com/spreadsheets/d/141h1o9ZTdpdTP9JuQT7QP5MK5UAFQ00bfymqVtyCHyU/edit?usp=sharing)

### CODE REVIEW

File: client/Packages/com.beamable.server/Editor/Usam/Editor/CodeService.cs [hunk 1]
```diff
@@ -215,7 +262,74 @@ private async Task<string> MigrateCommon()
 
 			return outputPath;
 		}
+		
+		private static List<IDescriptor> GetAllOldServices()
+		{
+			List<string> servicesToIgnore = new List<string>() {"CacheDependentMS"};
```
Comment (gabrielbeamable) [inline client/Packages/com.beamable.server/Editor/Usam/Editor/CodeService.cs:268]: Should we migrate this? Put as to ignore but I'm not sure what this even is about

**Comment by gabrielbeamable** [comment]
There is a bug with this PR that I found out, hold on...

File: client/Packages/com.beamable.server/Editor/Usam/Editor/CodeService.cs [hunk 2]
```diff
@@ -103,119 +107,187 @@ public async Promise Init()
...
+			var storageModel = MicroservicesDataModel.Instance.Storages.FirstOrDefault(s => s.Name == storageName);
+			var deps = MicroservicesDataModel.Instance.Services
+			                                 .Where(model => model.Dependencies.Any(s => s.Name == storageName)).Select(model => ServiceDefinitions.FirstOrDefault(d => d.BeamoId == model.Name))
+			                                 .ToList();
+			Debug.Log(storageModel);
+			Debug.Log(string.Join(", ", deps));
+			if (!Directory.Exists(path))
+			{
+				Debug.Log(storageName);	
+				await CreateStorage(storageName, deps, shouldInitialize: false);
```
Comment (cdhanna) [inline client/Packages/com.beamable.server/Editor/Usam/Editor/CodeService.cs:175]: we'd need to copy the src code included in the storage as well. Is that happening, and I'm not seeing it?

File: client/Packages/com.beamable.server/Editor/Usam/Editor/CodeService.cs (hunk 1 above)
Comment (cdhanna) [reply client/Packages/com.beamable.server/Editor/Usam/Editor/CodeService.cs:219]: honestly, I think we can delete it from our codebase wholesale.

**Review by cdhanna — CHANGES_REQUESTED** [review]
looks great- just one q

File: client/Packages/com.beamable.server/Editor/Usam/Editor/CodeService.cs (hunk 2 above)
Comment (gabrielbeamable) [reply client/Packages/com.beamable.server/Editor/Usam/Editor/CodeService.cs:175]: good catch, it isn't. I'm going to make that now

**Review by cdhanna — APPROVED** [review]

**Pull request merged**